### PR TITLE
Strobe shield now activates when attacked

### PR DESCRIPTION
**Pull request by Haevacht**
It activates the same way using a flash inhand does, seems to lack a stun but should give you an edge anyway.

Reason: Don't care if this doesn't happen in seige, this should make people use it in space, against unprepared tiders or cults. They can't hit what they can't seeeee!

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Nah

**Comment by ShadowDeath6** [comment]
>muder

**Comment by Haevacht** [comment]
I check for typos in code, not in commit messages, so hush you.

**Comment by GunHog** [comment]
I like it!

**Comment by SweaterKittens** [comment]
Can't you already use it in-hand for the same effect?

**Comment by Haevacht** [comment]
SK Yes, but then you have to swap off your weapon to blind the greytiders. It'll also burn out if someone spams it with attacks, like a regular flash.  Should make more people use it as it's more reliably going to blind people this way.

File: code/modules/assembly/flash.dm [hunk 1]
```diff
@@ -281,4 +281,8 @@
 		addtimer(CALLBACK(src, .proc/update_icon), 5)
 
 	if(holder)
-		holder.update_icon()
\ No newline at end of file
+		holder.update_icon()
+
+/obj/item/device/assembly/flash/shield/hit_reaction(obj/item/weapon/W, mob/user, params)
+	activate()
```
Comment (KorPhaeron) [inline code/modules/assembly/flash.dm]: Doesn't this mean it will activate even if the hit isn't blocked?

Comment (Haevacht) [inline code/modules/assembly/flash.dm]: Yes. But I can't see a way to get if the attack was blocked from hit_reaction. This does give them a weakness in that someone with glasses can spam you and burn out your shield.

Comment (PKPenguin321) [inline code/modules/assembly/flash.dm]: I think you'd have to do some shenanigans in check_block or something similar

Comment (optimumtact) [inline code/modules/assembly/flash.dm]: Yes I think this actually quite a nice little counter

**Pull request merged**